### PR TITLE
Setup for docker environment

### LINKED ISSUE #300: Add docker image to dockerhub and autodeploy to Google Cloud with Kubernetes
In order to make Yaydoc into a stable service we need to set up the infrastructure. Docker can help to do this more easily. Please signup on docker.com and join the yaydoc team. 

Create a docker file with Kubernetes and deploy automatically to Google cloud.

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

* Added Dockerfile

Dockerhub deployment: https://hub.docker.com/r/ujjwalbhardwaj/yaydoc/
## Related Issue

Partly fixes #300 
## Types of changes

- [ ] Bug fix 
- [ ] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by imujjwal96** [comment]
@pri22296 @sch00lb0y @niranjan94  Please review.

**Comment by sch00lb0y** [comment]
@imujjwal96 provide sample deployment

**Comment by imujjwal96** [comment]
@sch00lb0y http://163.172.11.243:3004/ Here's a deployment using  the repository at https://hub.docker.com/r/ujjwalbhardwaj/yaydoc/ adding the Dockerfile

**Comment by imujjwal96** [comment]
@pri22296 What's your review of this?

**Comment by pri22296** [comment]
@imujjwal96 `502 Bad gateway` on signing in CI

**Comment by imujjwal96** [comment]
@pri22296 It was because of incorrect env variables. Please test: http://163.172.11.243:3006

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+WORKDIR /usr/src/app
+
+# Install app dependencies
+COPY package.json .
+
+RUN npm install
+
+# Update and install packages
+RUN apt-get update && \
+    apt-get install -y python python-dev python-pip python-virtualenv zip rsync && rm -rf /var/lib/apt/lists/*
```
Comment (yukiisbored) [inline Dockerfile]: Do this first, it'll help decreasing build times when someone modifies the dependencies.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+# Install python dependencies
+COPY requirements.txt /usr/src/app/
+RUN pip install -r /usr/src/app/requirements.txt
+
+# Install Heroku CLI
+RUN wget -qO- https://cli-assets.heroku.com/install-ubuntu.sh | sh
```
Comment (yukiisbored) [inline Dockerfile]: Why do we need this?

File: .dockerignore [hunk 3]
```diff
@@ -0,0 +1,4 @@
+node_modules
+npm-debug.log
+temp
+.env
```
Comment (yukiisbored) [inline .dockerignore]: Missing `\n` on EOL.

File: Dockerfile (hunk 2 above)
Comment (imujjwal96) [inline Dockerfile]: Heroku CLI is required to deploy generated documentation to Heroku

**Comment by pri22296** [comment]
@imujjwal96 Is the deployment down? I'm getting `503 Service unavailable`

**Comment by imujjwal96** [comment]
@pri22296 i changed the port! Test with the link https://github.com/fossasia/yaydoc/pull/353#issuecomment-320450568 here

**Comment by pri22296** [comment]
@imujjwal96 I think the code in the `ON_HEROKU` section isn't needed here too. Can you check once? Coz I'm still getting the Hashmissing error. Also the dependencies are getting installed on each build due to it.

https://github.com/fossasia/yaydoc/blob/2e3c7d90c2f864d181d32ed5d316c0b796d85acf/generate.sh#L59

**Comment by imujjwal96** [comment]
Should I make ON_HEROKU true for docker?

**Comment by pri22296** [comment]
Yeah, and if it works properly, rename it to something meaningful then as it would be no longer just for Heroku. Otherwise, it would increase our build times.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,26 @@
+FROM node:boron
+
+# Create app directory
+WORKDIR /usr/src/app
+
+# Install app dependencies
+COPY package.json .
+
+# Update and install packages
+RUN apt-get update && \
```
Comment (yukiisbored) [inline Dockerfile]: Do this instruction first, it'll decrease the amount of time needed to rebuild.

**Comment by imujjwal96** [comment]
@pri22296  Updated

**Comment by pri22296** [comment]
@imujjwal96 Update the `app.json` file?

**Comment by imujjwal96** [comment]
@pri22296 Thanks for reminding!

**Comment by imujjwal96** [comment]
@pri22296 Updated that. Anything else?

**Comment by pri22296** [comment]
@imujjwal96 Please check the codacy issues

**Comment by imujjwal96** [comment]
I've compared it with other repositories at FOSSASIA. They haven't added a maintainer in their Dockerfile. Should I? :smile:

**Comment by pri22296** [comment]
Ask the mentors?

**Comment by pri22296** [comment]
> Similar apt commands are run at open-event-orga without the given tag! I am not sure if adding that tag wouldn't effect the app. We might need those additional packages. If you feel there is a need, I can try and test that though.

Nah, no need. Looks good. Just resolve the conflicts

**Pull request merged**